### PR TITLE
added LED abstractions; not close to done yet; not checked for errors yet

### LINKED ISSUE #50: Indicator LEDs
Currently, we don't use our indicator LEDs. We should! To start off with, we could use them for:
- DMX status
- I2C status

Since we have 2.

### PR DESCRIPTION
**Pull request by mplewis**
Closes #50 when it's done.

Not done yet. Please take a look and triple-check my bitwise hackery.

### CODE REVIEW

File: LED.c [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+void LEDOff(char led)
+{
+    led = led % 6;
+    PORTB &= ~(0x01 << led);
+}
+
+void LEDOut(char leds)
+{
+    leds = leds % ((1 << 6) - 1);       // 0b1000000 - 1 = 0b011111; limit ledOut to modifying 6 LSBs
```
Comment (kevana) [inline LED.c:23]: Is there some reason we can't just use:

``` c
PORTB = (leds & 0b00111111); 
```

Makes sure we don't change non-LED pins on PORTB and it's a bit more succinct than using modulo.

File: LED.c [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    // TODO: make ledOut output the proper values to RB0 ... RB5
+}
+
+char LEDGetLeds(void)
+{
+    return PORTB && 0b00111111;         // return the values of the 6 LEDs on RB0 ... RB5
```
Comment (kevana) [inline LED.c:29]: I think you'll just want a single & here. && does a boolean comparison.

File: LED.c [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+char LEDGetLed(char ledNum)
+{
+    ledNum = ledNum % 6;                // limit: 0 <= led <= 5; only 6 LEDs on board
+    return PORTB && (0x01 <<  ledNum);  //
```
Comment (kevana) [inline LED.c:35]: Same here, & instead of &&.

File: LED.c (hunk 2 above)
Comment (mplewis) [inline LED.c:29]: You are correct! Thanks for checking.

**Comment by kevana** [comment]
RIP led_abstractions branch, you will be missed.

**Pull request closed without merging**